--- /etc/xdg/awesome/rc.lua	2013-09-27 22:23:54.000000000 +0200
+++ rc.lua	2014-04-21 22:16:22.773554088 +0200
@@ -82,7 +82,7 @@
 
 -- This is used later as the default terminal and editor to run.
 terminal = "xterm"
-editor = os.getenv("EDITOR") or os.getenv("VISUAL") or "vi"
+editor = os.getenv("EDITOR") or os.getenv("VISUAL") or "vim"
 editor_cmd = terminal .. " -e " .. editor
 
 freedesktop.utils.terminal = terminal
@@ -180,18 +180,20 @@
 mybattery = widget({ type = "textbox"})
 vicious.register(mybattery, function(format, warg)
   local args = vicious.widgets.bat(format, warg)
-  if args[2] < 50 then
+  if args[2] < 25 then
     args['{color}'] = 'red'
+  elseif args[2] < 50 then
+    args['{color}'] = 'yellow'
   else
-    args['{color}'] = 'green'
+    args['{color}'] = 'lightgreen'
   end
   return args
 end, '<span foreground="${color}">bat: $2% $3h</span>', 10, 'BAT0')
 
 -- Initialize widget
-mynetwidget = widget({ type = "textbox" })
+--mynetwidget = widget({ type = "textbox" })
 -- Register widget
-vicious.register(mynetwidget, vicious.widgets.net, "${eth0 down_kb} / ${eth0 up_kb}", 1)
+--vicious.register(mynetwidget, vicious.widgets.net, "${eth0 down_kb} / ${eth0 up_kb}", 1)
 
 -- wifi
 -- provides wireless information for a requested interface
@@ -203,15 +205,15 @@
 
 
 -- Weather widget
-myweatherwidget = widget({ type = "textbox" })
-weather_t = awful.tooltip({ objects = { myweatherwidget },})
-vicious.register(myweatherwidget, vicious.widgets.weather,
-                function (widget, args)
-                    weather_t:set_text("City: " .. args["{city}"] .."\nWind: " .. args["{windkmh}"] .. "km/h " .. args["{wind}"] .. "\nSky: " .. args["{sky}"] .. "\nHumidity: " .. args["{humid}"] .. "%")
-                    return args["{tempc}"] .. "C"
-                end, 1800, "EDDN")
-                --'1800': check every 30 minutes.
-                --'EDDN': Nuernberg ICAO code.
+-- myweatherwidget = widget({ type = "textbox" })
+-- weather_t = awful.tooltip({ objects = { myweatherwidget },})
+-- vicious.register(myweatherwidget, vicious.widgets.weather,
+--                 function (widget, args)
+--                     weather_t:set_text("City: " .. args["{city}"] .."\nWind: " .. args["{windkmh}"] .. "km/h " .. args["{wind}"] .. "\nSky: " .. args["{sky}"] .. "\nHumidity: " .. args["{humid}"] .. "%")
+--                     return args["{tempc}"] .. "C"
+--                 end, 1800, "EDDN")
+--                 --'1800': check every 30 minutes.
+--                 --'EDDN': Nuernberg ICAO code.
 
 
 -- Keyboard map indicator and changer
@@ -219,23 +221,23 @@
 -- default keyboard is us, second is german adapt to your needs
 --
 
-    kbdcfg = {}
-    kbdcfg.cmd = "setxkbmap"
-    kbdcfg.layout = { { "us", "" }, { "de", "" } }
-    kbdcfg.current = 1  -- us is our default layout
-    kbdcfg.widget = widget({ type = "textbox", align = "right" })
-    kbdcfg.widget.text = " " .. kbdcfg.layout[kbdcfg.current][1] .. " "
-    kbdcfg.switch = function ()
-       kbdcfg.current = kbdcfg.current % #(kbdcfg.layout) + 1
-       local t = kbdcfg.layout[kbdcfg.current]
-       kbdcfg.widget.text = " " .. t[1] .. " "
-       os.execute( kbdcfg.cmd .. " " .. t[1] .. " " .. t[2] )
-    end
-
-    -- Mouse bindings
-    kbdcfg.widget:buttons(awful.util.table.join(
-        awful.button({ }, 1, function () kbdcfg.switch() end)
-    ))
+    --   kbdcfg = {}
+    --   kbdcfg.cmd = "setxkbmap"
+    --   kbdcfg.layout = { { "us", "" }, { "de", "" } }
+    --   kbdcfg.current = 1  -- us is our default layout
+    --   kbdcfg.widget = widget({ type = "textbox", align = "right" })
+    --   kbdcfg.widget.text = " " .. kbdcfg.layout[kbdcfg.current][1] .. " "
+    --   kbdcfg.switch = function ()
+    --      kbdcfg.current = kbdcfg.current % #(kbdcfg.layout) + 1
+    --      local t = kbdcfg.layout[kbdcfg.current]
+    --      kbdcfg.widget.text = " " .. t[1] .. " "
+    --      os.execute( kbdcfg.cmd .. " " .. t[1] .. " " .. t[2] )
+    --   end
+
+    --   -- Mouse bindings
+    --   kbdcfg.widget:buttons(awful.util.table.join(
+    --       awful.button({ }, 1, function () kbdcfg.switch() end)
+    --   ))
 
 -- Create a systray
 mysystray = widget({ type = "systray" })
@@ -320,10 +322,10 @@
         separator,
         spacer,
 
-        kbdcfg.widget,
-        spacer,
-        separator,
-        spacer,
+        -- kbdcfg.widget,
+        -- spacer,
+        -- separator,
+        -- spacer,
 
         mycpuwidget,
         spacer,
@@ -332,23 +334,23 @@
 
         mybattery,
         spacer,
-        separator,
-        spacer,
+        -- separator,
+        -- spacer,
 
-        mynetwidget,
-        spacer,
-        separator,
-        spacer,
+        --mynetwidget,
+        --spacer,
+        --separator,
+        --spacer,
 
         -- wifi,
         -- spacer,
         -- separator,
         -- spacer,
 
-        myweatherwidget,
-        spacer,
-        separator,
-        spacer,
+        -- myweatherwidget,
+        -- spacer,
+        -- separator,
+        -- spacer,
         s == 1 and mysystray or nil,
         mytasklist[s],
         layout = awful.widget.layout.horizontal.rightleft
